usr/src/uts/common/fs/zfs/sys/zio_impl.h

```diff
@@ -108,37 +118,38 @@ enum zio_stage {
 	ZIO_STAGE_OPEN			= 1 << 0,	/* RWFCI */
 
 	ZIO_STAGE_READ_BP_INIT		= 1 << 1,	/* R---- */
-	ZIO_STAGE_WRITE_BP_INIT		= 1 << 2,	/* -W--- */
-	ZIO_STAGE_FREE_BP_INIT		= 1 << 3,	/* --F-- */
-	ZIO_STAGE_ISSUE_ASYNC		= 1 << 4,	/* RWF-- */
+	ZIO_STAGE_FREE_BP_INIT		= 1 << 2,	/* --F-- */
+	ZIO_STAGE_ISSUE_ASYNC		= 1 << 3,	/* RWF-- */
+	ZIO_STAGE_WRITE_BP_INIT		= 1 << 4,	/* -W--- */
```

Comment:
illumos and ZoL both have the commit for illumos bug 7090, and have the order you quoted:
```
WRITE_BP_INIT
FREE_BP_INIT
ISSUE_ASYNC
WRITE_COMPRESS
```
This order isn't changed by the ZoL encryption PR.  The ZoL encryption PR adds a new stage, ENCRYPT, after WRITE_COMPRESS.  The illumos encryption PR should do the same.